pages/_app.js

```diff
@@ -23,30 +68,17 @@ export default function App({ Component, pageProps }) {
         return event;
       });
       setNewsEvents(updatedEventsArray);
+      setAllEvents(updatedEventsArray);
     }
-  }, [events, alarmEvents]);
-
-  function getEachDaysEvents(events) {
-    const weekdayEvents = {};
-    weekdayEvents.monday = events.filter((event) => {
-      return event.weekday === "Monday";
-    });
-    weekdayEvents.tuesday = events.filter((event) => {
-      return event.weekday === "Tuesday";
-    });
-    weekdayEvents.wednesday = events.filter((event) => {
-      return event.weekday === "Wednesday";
-    });
-    weekdayEvents.thursday = events.filter((event) => {
-      return event.weekday === "Thursday";
-    });
-    weekdayEvents.friday = events.filter((event) => {
-      return event.weekday === "Friday";
-    });
-
-    return weekdayEvents;
-  }
-
+  }, [events, alarmEvents, settings]);
+  useEffect(() => {
+    if (settings.flagsTurnedOn) {
+      const preferredCurrenciesOnly = allEvents.filter(
+        (event) => settings.countryFlags[event.country]
+      );
+      setNewsEvents(preferredCurrenciesOnly);
+    }
+  }, [allEvents, settings]);
```

Comment:
Why is this in an effect? You can filter without an effect, even if it is only filtered conditionally.

```js
const eventsToShow = settings.flagsTurnedOn
  ? allEvents.filter((event) => settings.countryFlags[event.country])
  : allEvents;
```

Reply:
You also do not need two states to do so (see above)

Reply:
thanks a lot for this piece of advice 👯